eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -312,7 +312,7 @@ jobs:
         condition: and(succeeded(), or(ne(variables['TestFromSource'],'true'), eq(variables['ShouldRunSourceTests'],'true')))
 
       - task: PublishTestResults@2
-        condition: and(always(), or(ne(variables['TestFromSource'],'true'), eq(variables['ShouldRunSourceTests'],'true')))
+        condition: and(succeededOrFailed(), or(ne(variables['TestFromSource'],'true'), eq(variables['ShouldRunSourceTests'],'true')))
```

Comment:
I would prefer to keep `PublishTestResults` as `always()`.  This task should only take a few seconds, and when cancelling a build (say due to a hung test) it's valuable to publish the test results we do have so far.

Reply:
Using ```always()``` will result in the job ignoring cancellation requests. Perhaps ```succeededOrFailed()``` might be more appropriate?

Reply:
I assume you mean the `PublishTestResults` **task** will ignore cancellation, not the whole `Test` job, right?

Reply:
If the job has started then it'll run ```PublishTestResults``` with ```always()```. If the job hasn't started then the task won't run. Its probably not a big deal, if the job hasn't started then it won't run which will help with recovery, and provided we always put a ```succeeded()``` or ```succeededOrFailed()``` on preceding steps where a condition is used then it probably won't be a big issue.

I think you've convinced me ;)

Reply:
Sounds good.  If this does end up causing bigger problems with cancellation, I agree it's not worth the benefit and should be changed to `succeededOrFailed()`.